sphinx/writers/latex.py

```diff
@@ -1959,6 +1959,10 @@ def add_target(id):
             if id.startswith('index-'):
                 return
 
+            # equations also need no extra blank line nor hypertarget
+            if id.startswith('equation-'):
```

Comment:
math elements are provided as extension. According to basis, LaTeX writer should not know about `equation-` label.
Could you add a TODO here please?

Reply:
OK, I will. Aren't you planning to merge mathbase into Sphinx-core ?

Reply:
Yes! see #4333